File: TUnit.TestProject.FSharp/DeepNestedDependencyConflict.fs
namespace TUnit.TestProject.FSharp

open TUnit.Core
open System.Diagnostics.CodeAnalysis

// Equivalent of DeepNestedDependencyConflict.cs

[<SuppressMessage("Usage", "TUnit0033:Conflicting DependsOn attributes")>]
type DeepNestedDependencyConflict() =
    [<Test>]
    [<DependsOn("Test2")>]
    member _.Test1() = ()

    [<Test>]
    [<DependsOn("Test3")>]
    member _.Test2() = ()

    [<Test>]
    [<DependsOn("Test4")>]
    member _.Test3() = ()

    [<Test>]
    [<DependsOn("Test5")>]
    member _.Test4() = ()

    [<Test>]
    [<DependsOn("Test6")>]
    member _.Test5() = ()

    [<Test>]
    [<DependsOn("Test7")>]
    member _.Test6() = ()

    [<Test>]
    [<DependsOn("Test8")>]
    member _.Test7() = ()

    [<Test>]
    [<DependsOn("Test9")>]
    member _.Test8() = ()

    [<Test>]
    [<DependsOn("Test10")>]
    member _.Test9() = ()

    [<Test>]
    [<DependsOn("Test9")>]
    member _.Test10() = ()

File: tests/TUnit.TestProject/DeepNestedDependencyConflict.cs
using System.Diagnostics.CodeAnalysis;
using TUnit.TestProject.Attributes;

namespace TUnit.TestProject;

[EngineTest(ExpectedResult.Failure)]
[UnconditionalSuppressMessage("Usage", "TUnit0033:Conflicting DependsOn attributes")]
public class DeepNestedDependencyConflict
{
    [Test]
    [DependsOn(nameof(Test2))]
    public void Test1()
    {
    }

    [Test]
    [DependsOn(nameof(Test3))]
    public void Test2()
    {
    }

    [Test]
    [DependsOn(nameof(Test4))]
    public void Test3()
    {
    }

    [Test]
    [DependsOn(nameof(Test5))]
    public void Test4()
    {
    }

    [Test]
    [DependsOn(nameof(Test6))]
    public void Test5()
    {
    }

    [Test]
    [DependsOn(nameof(Test7))]
    public void Test6()
    {
    }

    [Test]
    [DependsOn(nameof(Test8))]
    public void Test7()
    {
    }

    [Test]
    [DependsOn(nameof(Test9))]
    public void Test8()
    {
    }

    [Test]
    [DependsOn(nameof(Test10))]
    public void Test9()
    {
    }

    [Test]
    [DependsOn(nameof(Test9))]
    public void Test10()
    {
    }
}
